client-ui/src/app/components/modal/modal-expression/modal-expression.component.less

```diff
@@ -14,76 +14,160 @@
-      padding-left: 5px;
-    }
-  
-    /deep/ .tree-node-children {
-      padding-left: 15px;
-    }
-  
-    /deep/ .tree-node {
-  
-      padding-left: 10px;
-      border-left: 1px dotted #23527c;
-  
-    }
-  
-    /deep/ .node-content-wrapper {
-      ::before {
-        left: -10px;
+
+    tree-node > div {
+
+      > .node-wrapper {
+        margin-left: 24px;
+
+        > .node-content-wrapper {
+          margin-left: 4px;
+
+          tree-node-content {
+            margin-left: 30px;
+
+            .node-title {
+              color: red
+            }
+          }
+        }
+      }
+
+      &.tree-node-leaf > .node-wrapper {
+        margin-left: 0
+      }
+
+      &::before {
+        content: "";
+        position: absolute;
         border-bottom: 1px dotted #23527c;
-        width: 10px;
+        width: 7px;
+        margin-top: 12px;
+        left: 7px
+      }
+
+      .toggle-children-wrapper {
+        width: 13px;
+        height: 13px;
+        border: 1px solid #23527c;
+        position: absolute;
+        left: 15px;
+        margin-top: 5px;
+        margin-left: 0;
+        display: inline-block;
+        background-color: #fff;
+        z-index: 1;
+
+        &::before {
+          content: "";
+          display: inline-block;
+          width: 7px;
+          border-top: 1px solid #23527c;
+          position: absolute;
+          top: 5px;
+          left: 2px
+        }
+
+        &.toggle-children-wrapper-collapsed::after {
+          content: "";
+          display: inline-block;
+          height: 7px;
+          border-left: 1px solid #23527c;
+          position: absolute;
+          top: 2px;
+          left: 5px
+        }
+
+        .toggle-children {
+          display: none
+        }
+      }
+
+      .node-content-wrapper {
+        margin-left: 4px;
+
+        tree-node-content {
+          margin-left: 32px;
+
+          .node-title {
+            &::before,
+            &::after {
+              margin: 4px;
+            }
+
+            &.expr-Generic::before,
+            &.expr-GenericLeaf::before {
+              content: "f(x)";
+              color: dodgerblue;
+            }
+
+            &.expr-Union::before {
+              content: "⋃";
+              color: dodgerblue;
+            }
+
+            &.expr-UDF::before {
+              content: "UDF";
+              color: dodgerblue;
+            }
+
+            &.expr-Binary {
+              font-weight: bold;
+              color: dodgerblue;
+            }
+
+            &.expr-Alias,
+            &.expr-AttrRef {
+              font-weight: bold;
+
+              &::before {
+                -moz-osx-font-smoothing: grayscale;
+                -webkit-font-smoothing: antialiased;
+                display: inline-block;
+                font-style: normal;
+                font-variant: normal;
+                text-rendering: auto;
+                line-height: 1;
+                font-family: "Font Awesome 5 Free";
```

Comment:
You will have here an error. There should be at leas one generic font defined, e.g. like so: font-family: "Font Awesome 5 Free", sans-serif;

Reply:
What's the point? This is a copy from the FontAwesome CSS.

Reply:
The point is that there is a rule in Sonar, which requires definition of some default font which will be used in case Font Awesome font will not be found. 

Agree, it makes no sense in that case. But you have to fix it or disable that rule in Sonar.

Reply:
I think we'll rather deal with Sonar. The code clarity and understandability is more important than silly rules.